fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -53,6 +53,54 @@ public FabricItemSettings customDamage(CustomDamageHandler handler) {
 		return this;
 	}
 
+	/**
+	 * Sets the stack-aware recipe remainder provider of the item.
+	 * @see RecipeRemainderProvider
+	 */
+	public FabricItemSettings recipeRemainder(RecipeRemainderProvider provider) {
+		FabricItemInternals.computeExtraData(this).recipeRemainderProvider(provider);
+		return this;
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by 1 every time it is used in crafting.
+	 */
+	public FabricItemSettings damageIfUsedInCrafting() {
+		return this.damageIfUsedInCrafting(1);
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to return the item itself.
+	 */
+	public FabricItemSettings returnSelfInCrafting() {
+		return this.damageIfUsedInCrafting(0);
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by a certain amount every time it is used in crafting.
+	 *
+	 * @param by the amount
+	 */
+	public FabricItemSettings damageIfUsedInCrafting(int by) {
+		return this.recipeRemainder((original, inventory, type, world, pos) -> {
+			if (by == 0) {
```

Comment:
If you can move this `by == 0` check outside of the lambda creation (i.e. create different lambdas when by is different values), we can have a performance boost for the `by == 0` special case lambdas.